fawltydeps/main.py

```diff
@@ -211,53 +211,74 @@ def build_parser() -> argparse.ArgumentParser:
     return parser
 
 
-def main() -> int:
-    """Command-line entry point."""
-    parser = build_parser()
-    args = parser.parse_args()
-    settings = Settings.config(config_file=args.config_file).create(args)
+def assign_exit_code(analysis: Analysis) -> int:
+    """
+    Assign exit code based on the analysis results.
 
-    logging.basicConfig(level=logging.WARNING - 10 * settings.verbosity)
-
-    if args.generate_toml_config:
-        print_toml_config(settings, sys.stdout)
-        return 0
+    Exit codes:
+    0 - success, no problems found
+    1 - an exception propagates (this should not happen)
+    2 - command-line parsing error (generated in 'main')
+    3 - undeclared dependencies found
+    4 - unused dependencies found
+    """
+    if analysis.is_enabled(Action.REPORT_UNDECLARED) and analysis.undeclared_deps:
+        return 3
+    if analysis.is_enabled(Action.REPORT_UNUSED) and analysis.unused_deps:
+        return 4
 
-    try:
-        analysis = Analysis.create(settings)
-    except UnparseablePathException as exc:
-        return parser.error(exc.msg)  # exit code 2
+    return 0
 
-    # Exit codes:
-    # 0 - success, no problems found
-    # 1 - an exception propagates (this should not happen)
-    # 2 - command-line parsing error (see above)
-    # 3 - undeclared dependencies found
-    # 4 - unused dependencies found
-    exit_code = 0
-    if analysis.is_enabled(Action.REPORT_UNDECLARED) and analysis.undeclared_deps:
-        exit_code = 3
-    elif analysis.is_enabled(Action.REPORT_UNUSED) and analysis.unused_deps:
-        exit_code = 4
 
+def print_output(
+    analysis: Analysis,
+    exit_code: int,
+    stdout: TextIO = sys.stdout,
+) -> None:
+    """Print the summary message for CLI run."""
```

Comment:
Nit: Docstring no longer correspond to function name:

```suggestion
    """Print the output of the given 'analysis' to 'stdout'."""
```